server/service/src/rnr_form/finalise.rs

```diff
@@ -36,15 +36,28 @@ pub fn finalise_rnr_form(
     let rnr_form = ctx
         .connection
         .transaction_sync(|connection| {
+            let requisition_repo = RequisitionRowRepository::new(&ctx.connection);
+            let requisition_line_repo = RequisitionLineRowRepository::new(&ctx.connection);
+            let rnr_form_repo = RnRFormRowRepository::new(connection);
+            let rnr_form_line_repo = RnRFormLineRowRepository::new(&ctx.connection);
+
             let rnr_form = validate(ctx, store_id, &input)?;
 
-            // Create an internal order based on the RnR form
-            let requisition_id = create_internal_order(ctx, rnr_form.clone())?;
+            let GenerateResult {
+                requisition_row,
+                finalised_rnr_form,
+                rnr_form_line_and_requisition_lines,
+            } = generate(ctx, rnr_form)?;
 
-            let finalised_form = generate(rnr_form, requisition_id);
-            let rnr_form_repo = RnRFormRowRepository::new(connection);
+            requisition_repo.upsert_one(&requisition_row)?;
+            rnr_form_repo.upsert_one(&finalised_rnr_form)?;
 
-            rnr_form_repo.upsert_one(&finalised_form)?;
+            for (rnr_form_line_id, requisition_line) in rnr_form_line_and_requisition_lines {
+                requisition_line_repo.upsert_one(&requisition_line)?;
+
+                rnr_form_line_repo
+                    .update_requisition_line_id(&rnr_form_line_id, &requisition_line.id)?;
```

Comment:
Sorry, did a bit of rejigging in here so all the inserts/updates happened at the top level - this is the actual change, adding requisition_line_id to the rnr_form_lines